PiwikTracker/Event.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  Event.swift
+//  PiwikTracker
+//
+//  Created by Cornelius Horstmann on 21.12.16.
+//  Copyright © 2016 PIWIK. All rights reserved.
+//
+
+import Foundation
+
+/// Represents an event of any kind.
+///
+/// - Note: Should we store the resolution in the Event (cleaner) or add it before transmission (smaller)?
```

Comment:
Hm. Unsure about that. In the end the useragent represents the piece of code responsible for transmitting the data. Per default I would like it to be defined by the transmitting object/class/struct.

The resolution is independent of the transmitter and somehow it feels wrong to add it there. 

Idea: What if the Event has a reference to a "Context" object that is a reference to a Singleton (Context.current or such). This way we don't have to serialize/de-serialize it for storing it in CoreData and such but it is still a property of the Event.